File: backend/app/schemas/review.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreateRequest(BaseModel):
    tutor_id: uuid.UUID
    rating: int = Field(ge=1, le=5)
    text: str = Field(min_length=10, max_length=5000)
    sessions_attended: int = Field(ge=1)
    approximate_duration_weeks: int = Field(ge=1)


class ReviewUpdateRequest(BaseModel):
    rating: int | None = Field(None, ge=1, le=5)
    text: str | None = Field(None, min_length=10, max_length=5000)


class ReviewResponse(BaseModel):
    id: uuid.UUID
    tutor_id: uuid.UUID
    rating: int
    text: str
    reviewer_first_name: str | None = None
    reviewer_last_name: str | None = None
    sessions_attended: int | None = None
    approximate_duration_weeks: int | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: mobile/src/types/review.ts
export interface ReviewCreateRequest {
  tutor_id: string;
  rating: number;
  text: string;
  sessions_attended: number;
  approximate_duration_weeks: number;
}

export interface ReviewUpdateRequest {
  rating?: number;
  text?: string;
}

export interface ReviewResponse {
  id: string;
  tutor_id: string;
  rating: number;
  text: string;
  reviewer_first_name: string | null;
  reviewer_last_name: string | null;
  sessions_attended: number | null;
  approximate_duration_weeks: number | null;
  created_at: string;
  updated_at: string;
}
